packages/contracts-bedrock/scripts/upgrades/holocene/scripts/sys-cfg-bundle.sh

```diff
@@ -22,14 +30,72 @@ L1_CHAIN_ID=$(cast chain-id)
 # Copy the bundle template
 cp ./templates/sys_cfg_upgrade_bundle_template.json "$BUNDLE_PATH"
 
-# Tx 1: Upgrade SystemConfigProxy implementation
-TX_1_PAYLOAD=$(cast calldata "upgrade(address,address)" "$SYSTEM_CONFIG_PROXY_ADDR" "$SYSTEM_CONFIG_IMPL")
+# We need to re-generate the SystemConfig initialization call
+# We want to use the exact same values that the SystemConfig is already using, apart from baseFeeScalar and blobBaseFeeScalar.
+# Start with values we can just read off:
+OWNER=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "owner()")
+SCALAR=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "scalar()")
+BATCHER_HASH=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "batcherHash()")
+GAS_LIMIT=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "gasLimit()")
+UNSAFE_BLOCK_SIGNER=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "unsafeBlockSigner()")
+RESOURCE_CONFIG=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "resourceConfig()")
+BATCH_INBOX=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "batchInbox()")
+GAS_PAYING_TOKEN=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "gasPayingToken()(address)")
+L1_CROSS_DOMAIN_MESSENGER_PROXY=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "l1CrossDomainMessenger()(address)")
+L1_STANDARD_BRIDGE_PROXY=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "l1StandardBridge()(address)")
+L1_ERC721_BRIDGE_PROXY=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "l1ERC721Bridge()(address)")
+DISPUTE_GAME_FACTORY_PROXY=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "disputeGameFactory()(address)")
+OPTIMISM_PORTAL_PROXY=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "optimismPortal()(address)")
+OPTIMISM_MINTABLE_ERC20_FACTORY_PROXY=$(cast call "$SYSTEM_CONFIG_PROXY_ADDR" "optimismMintableERC20Factory()(address)")
+
+
+# Decode base fee scalar and blob base fee scalar from scalar value:
```

Comment:
This is insane and there should be a json output mode so that its easy to tell that it just works using jq

Reply:
Agreed. I wanted to reuse code that has tests and so on instead of rolling my own decoder. And didn't have the time yet to modify the tool to make the output easier to parse.